pkgs/development/python-modules/opensfm/default.nix

```diff
@@ -0,0 +1,106 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cmake
+, opencv
+, opencv4
+, ceres-solver
+, suitesparse
+, metis
+, eigen
+, pkg-config
+, pybind11
+, numpy
+, pyyaml
+, lapack
+, gtest
+, gflags
+, glog
+, pytestCheckHook
+, networkx
+, pillow
+, exifread
+, gpxpy
+, pyproj
+, python-dateutil
+, joblib
+, repoze_lru
+, xmltodict
+, cloudpickle
+, scipy
+}:
+
+let
+  pname = "OpenSfM";
+  version = "0.5.1";
+  src = fetchFromGitHub {
+    owner = "mapillary";
+    repo = "OpenSfM";
+    rev = "v${version}";
+    sha256 = "sha256-AU3n3AVgNWeRBFt5hnuzDVU4WEWhI9mmKUuARnFBY4c=";
+    fetchSubmodules = true;
+  };
+  opensfm =
+    buildPythonPackage {
+      inherit pname version src;
+      nativeBuildInputs = [ cmake pkg-config ];
+      buildInputs = [
+        opencv
+        ceres-solver.dev
+        suitesparse
+        metis
+        eigen
+        lapack
+        gflags
+        gtest
+        glog
+      ];
+      propagatedBuildInputs = [
+        numpy
+        scipy
+        pyyaml
+        opencv4
+        networkx
+        pillow
+        exifread
+        gpxpy
+        pyproj
+        python-dateutil
+        joblib
+        repoze_lru
+        xmltodict
+        cloudpickle
+      ];
+      checkInputs = [ pytestCheckHook ];
+      dontUseCmakeBuildDir = true;
+      dontUseCmakeConfigure = true;
+      cmakeFlags = [
+        "-Bcmake_build"
+        "-Sopensfm/src"
+      ];
+      patches = [ ./fix-cmake.patch ];
+      postPatch = ''
+        rm opensfm/src/cmake/FindEigen.cmake
+        rm opensfm/src/cmake/FindCeres.cmake
+        rm opensfm/src/cmake/FindGlog.cmake
+        rm opensfm/src/cmake/FindGflags.cmake
+        rm -rf opensfm/src/third_party/gtest
+      '';
+      disabledTestPaths = [
+        "opensfm/test/test_pairs_selection.py"
+      ];
+      disabledTests = [
+        "test_run_all"
+      ];
+      outputs = [ "out" "viewer" ];
+      postInstall = ''
+        cp --no-preserve=mode -r ./viewer/ $viewer
+      '';
+      meta = {
+        maintainers = [ lib.maintainers.SomeoneSerge ];
+        license = lib.licenses.bsd2;
+        description = "Open source Structure-from-Motion pipeline";
+      };
+    };
+in
```

Comment:
I don't think we need this let in construct for everything. It over complicated things a bit.

Reply:
Yes, I'll remove it. I was playing with splitting the binary part of the package before submission and left out the `let ... in`